### PR TITLE
add documentations for connection test data

### PR DESCRIPTION
**Pull request by YannanGao-gs**
add documentations for connection test data
Screenshot 2022-11-11 at 12 25 14 PM

### CODE REVIEW

File: docs/tutorials/studio-tutorial.md [hunk 1]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
 
 ## Create a Service Test 
 
+### Create connections test data
+Click the `generate` button to automatically generate test data if possible. Otherwise, please manually create test data.
+#### Data Type -- Relational
+Add relational data tables based on databases to which the service's mapping connects. It's unnecessary to provide all columns with test data. 
+The columns used in the service, however, require test data. Besides, the primary keys have to be distinct.
```
Comment (MauricioUyaguari) [inline docs/tutorials/studio-tutorial.md:579]: Lets add something like: 
Test data for relational is defined through csv values for each table. You must specify the schema name, table name and the corresponding csv alues.

File: docs/tutorials/studio-tutorial.md [hunk 2]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
...
+![Create connections test data - Relational](../assets/create-a-relational-service-connection-test-data.gif)
+#### Data Type -- Data element
+One `Data element` could be used among different tests.
```
Comment (MauricioUyaguari) [inline docs/tutorials/studio-tutorial.md:582]: `Data Elements` are top level elements that define data. They are useful for sharing test data among different tests.

File: docs/tutorials/studio-tutorial.md [hunk 3]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
...
+![Create connections test data - Data Element](../assets/create-a-service-connection-test-with-data-element.gif)
 ### Relational
 
 ![Create a service test - Relational](../assets/create-a-service-test-relational.gif)
```
Comment (MauricioUyaguari) [inline docs/tutorials/studio-tutorial.md:586]: Ca we add entires for External Format (M2M) and service store test data?

File: docs/tutorials/studio-tutorial.md [hunk 4]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
 
 ## Create a Service Test 
 
+### Create connections test data
```
Comment (MauricioUyaguari) [inline docs/tutorials/studio-tutorial.md:575]: I think we should move the Test Data to its own section in the docs. we can reference it in the service test. But they will be shared along for mapping tests as well. Please reorganize

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: docs/tutorials/studio-tutorial.md (hunk 4 above)
Comment (hardikmaheshwari) [reply docs/tutorials/studio-tutorial.md:575]: +1
Test Data will be used across the platform with multiple diff elements

File: docs/tutorials/studio-tutorial.md [hunk 5]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
...
+Click the `generate` button to automatically generate test data if possible. Otherwise, please manually create test data.
```
Comment (hardikmaheshwari) [inline docs/tutorials/studio-tutorial.md:576]: Maybe we should also mention something around when should a particular Data Type be used. 
Current strategy for defining subtypes is 
1 for each store types
1 for externalFormats
1 for reference

File: docs/tutorials/studio-tutorial.md [hunk 6]
```diff
@@ -572,6 +572,15 @@ This panel lets you define tests for the services you write by specifying test d
...
+#### Data Type -- Relational
+Add relational data tables based on databases to which the service's mapping connects. It's unnecessary to provide all columns with test data.
```
Comment (hardikmaheshwari) [inline docs/tutorials/studio-tutorial.md:578]: Hmm should we make this a bit more user centric, what do you think
maybe something along what/when/how should it be used?
what - A type of embedded data used to specify test data for relational data sources
when - To be used when user is sourcing data from relational db's
how - ...

File: docs/tutorials/studio-tutorial.md (hunk 2 above)
Comment (hardikmaheshwari) [reply docs/tutorials/studio-tutorial.md:582]: @MauricioUyaguari can we use 
`Data Elements` are packageable elements that wraps test data for store/external Formats. They are useful ..

File: docs/tutorials/studio-tutorial.md (hunk 3 above)
Comment (hardikmaheshwari) [reply docs/tutorials/studio-tutorial.md:586]: +1

File: docs/tutorials/studio-tutorial.md [hunk 7]
```diff
@@ -218,8 +218,26 @@ Create a relationship between the **Person** and **Firm** classes.
 
 ![Add a super type](../assets/add-non-primitive-data-type.gif)
 
-## Create a Diagram
+## Create a Data Element
+Data Elements are top level elements that define test data for store/external formats. They are useful for sharing test data among different tests.
+### Relational
+Test data for relational is defined through csv values for each table, which is to be used when a user is sourcing data from relational databases. The name of the schema, the table, and the related CSV values must all be specified. Besides, the primary keys have to be distinct.
+![relational data element](../assets/relational_db.png)
```
Comment (MauricioUyaguari) [inline docs/tutorials/studio-tutorial.md:225]: please update names and description 
data-element-relational-{description}

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**